src/Node/URL.js

```diff
@@ -1,10 +1,58 @@
-import url from "url";
-import queryString from "querystring";
-export { parse, format } from "url";
+import url from "node:url";
 
-export function resolve(from) {
-  return to => url.resolve(from, to);
-}
+export const newImpl = (input) => new url.URL(input);
+export const newRelativeImpl = (input, base) => new url.URL(input, base);
+export const pathToFileURLImpl = (path) => url.pathToFileURL(path);
+export const hashImpl = (u) => u.hash;
+export const setHashImpl = (h, u) => {
+  u.hash = h;
+};
+export const hostImpl = (url) => url.host;
+export const setHostImpl = (val, u) => {
+  u.host = val;
+};
+export const hostnameImpl = (u) => u.hostname;
+export const setHostnameImpl = (val, u) => {
+  u.hostname = val;
+};
+export const uneffectfulHref = (u) => u.href;
```

Comment:
Maybe this should be hrefPure? I think that’s a more common naming convention than “uneffectful”

Reply:
I like that name better. I'll send a PR.